crates/sui-sdk/src/crypto.rs

```diff
@@ -138,69 +175,8 @@ impl FileBasedKeystore {
     }
 }
 
-pub struct SuiKeystore(Box<dyn AccountKeystore>);
```

Comment:
the goal of this PR is to remove the pattern here that we box a dynamic trait. the limitation here is when when introducing generic types in a method with Intent Signing to AccountKeyStore, Rust will not compile due to unknown size.